File: string/128. Longest Consecutive Sequence.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int longestConsecutive(int* nums, int numsSize) {
    if (numsSize == 0) {
        return 0;
    }

    qsort(nums, numsSize, sizeof(int), compare);

    int cnt = 1;
    int maxi = 0;

    for (int i = 1; i < numsSize; i++) {
        if (nums[i] != nums[i - 1]) {
            if (nums[i] == nums[i - 1] + 1) {
                cnt++;
            } else {
                maxi = (maxi > cnt) ? maxi : cnt;
                cnt = 1;
            }
        }
    }

    return (maxi > cnt) ? maxi : cnt;
}

int main() {
    int nums[] = {100, 4, 200, 1, 3, 2};
    int numsSize = sizeof(nums) / sizeof(nums[0]);
    printf("Longest consecutive sequence length: %d\n", longestConsecutive(nums, numsSize));
    return 0;
}

File: string/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        mp=defaultdict(list)
        bl=defaultdict(bool)
        mx=0
        for i in nums:
            if bl[i]:
                continue 
            bl[i]=True
            l,r=i,i
            if mp[i+1]:
                r=mp[i+1][0]
            if mp[i-1]:
                l=mp[i-1][1]
            mp[r]=[r,l]
            mp[l]=[r,l]
            mx=max(mx,r-l+1)
        return mx
    
# Version 2 - C
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        n = len(nums)
        
        if n == 0:
            return 0

        nums.sort()

        cnt = 1
        maxi = 0

        for i in range(1, n):
            if nums[i] != nums[i - 1]:
                if nums[i] == nums[i - 1] + 1:
                    cnt += 1
                else:
                    maxi = max(maxi, cnt)
                    cnt = 1

        return max(maxi, cnt)